src/dataset-builder/dataset-builder-types.ts

```diff
@@ -53,14 +54,51 @@ export interface DomainCriteriaSelectorState extends DatasetBuilderState {
   readonly cohort: Cohort;
   readonly criteriaGroup: CriteriaGroup;
   readonly domainOption: DomainOption;
+  readonly cart: Concept[];
+  readonly cancelState: AnyDatasetBuilderState;
 }
 
 export const domainCriteriaSelectorState = {
-  new: (cohort: Cohort, criteriaGroup: CriteriaGroup, domainOption: DomainOption): DomainCriteriaSelectorState => ({
+  new: (
+    cohort: Cohort,
+    criteriaGroup: CriteriaGroup,
+    domainOption: DomainOption,
+    cart: Concept[],
+    cancelState: AnyDatasetBuilderState
+  ): DomainCriteriaSelectorState => ({
     mode: 'domain-criteria-selector',
     cohort,
     criteriaGroup,
     domainOption,
+    cart,
+    cancelState,
+  }),
+};
+
+export interface DomainCriteriaSearchState extends DatasetBuilderState {
+  mode: 'domain-criteria-search';
+
+  readonly cohort: Cohort;
+  readonly criteriaGroup: CriteriaGroup;
+  readonly domainOption: DomainOption;
+  readonly cart: Concept[];
+  readonly searchText: string;
+}
+
+export const domainCriteriaSearchState = {
+  new: (
+    cohort: Cohort,
+    criteriaGroup: CriteriaGroup,
+    domainOption: DomainOption,
+    cart: Concept[],
```

Comment:
Would it be worth giving default values for cart and searchTerm? I feel like that might save a lot of repeated code/boilerplate

Reply:
Sure, I can do this, although there is currently only two uses of `new()`, and only one of them passes in an empty cart and search string.

Reply:
default arguments added to `domainCriteriaSearchState`